handlers/secrets.go

```diff
@@ -75,3 +81,86 @@ func makeSecretsArray(c *client.Client, secretNames []string) ([]*swarm.SecretRe
 
 	return values, nil
 }
+
+func secretGetHandler(c *client.Client, w http.ResponseWriter, r *http.Request) {
+	secrets, err := c.SecretList(context.Background(), types.SecretListOptions{})
+
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+
+		return
+	}
+
+	var results []requests.Secret
+
+	for _, s := range secrets {
+		results = append(results, requests.Secret{Name: s.Spec.Name})
+	}
+
+	bs, marshalErr := json.Marshal(results)
+
+	if marshalErr != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+
+		return
+	}
+
+	_, writeErr := w.Write(bs)
+
+	if writeErr != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+}
+
+func secretPostHandler(c *client.Client, w http.ResponseWriter, r *http.Request) {
+	bs, err := ioutil.ReadAll(r.Body)
+
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+
+		return
+	}
+
+	var secret requests.Secret
+
+	unmarshalErr := json.Unmarshal(bs, &secret)
+
+	if unmarshalErr != nil {
+		w.WriteHeader(http.StatusBadRequest)
+
+		return
+	}
+
+	_, createSecretErr := c.SecretCreate(context.Background(), swarm.SecretSpec{
+		Annotations: swarm.Annotations{
+			Name: secret.Name,
+		},
+		Data: []byte(secret.Value),
+	})
+
+	if createSecretErr != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+
+		return
+	}
+
+	w.WriteHeader(http.StatusCreated)
+
+}
+
+func MakeSecretsHandler(c *client.Client) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		switch r.Method {
+		case http.MethodPost:
+			secretPostHandler(c, w, r)
+
```

Comment:
I suppose the empty lines may come from the IDE settings. Can you please delete them for better readability?

Reply:
The case is that it's not from IDE, but I think that it's easier to read like it is now (with empty lines). But I can remove them if it's more readable for you. @alexellis @LucasRoesler wdyt?

Reply:
I think almost all go code I have ever read puts error handling immediately on the next line without any spacing (although i could easily be wrong about that). When it is spaced out i read it as having a lot more logical blocks than there actually are.

Reply:
I changed that regarding yours comments